lib/pulsar/client_configuration.rb

```diff
@@ -41,18 +45,83 @@ def self.from_environment(config={})
       if ENV.key?('PULSAR_AUTH_TOKEN')
         env_config[:authentication_token] = ENV['PULSAR_AUTH_TOKEN']
       end
+      if ENV.key?('PULSAR_CLIENT_CONF')
+        env_config.merge!(read_from_client_conf(ENV['PULSAR_CLIENT_CONF']))
+      end
       self.from(env_config.merge(config))
     end
 
+    # Load configuration from PULSAR_CLIENT_CONF
+    # refer to (see https://github.com/apache/pulsar/blob/master/conf/client.conf)
+    def self.read_from_client_conf(pulsar_client_conf_file)
+      client_config = {}
+      # Read the client config as a Ruby Hashmap
+      pulsar_config = read_config(pulsar_client_conf_file)
+      # Attempt to read as many configuration as possible from the client configuration
+
+      # If a TLS certificate had been given, use it
+      if pulsar_config.has_key? 'tlsTrustCertsFilePath'
+        client_config[:use_tls] = true
+        client_config[:tls_trust_certs_file_path] = pulsar_config['tlsTrustCertsFilePath']
+      end
+      # If 'TLS enable hostname verification' is false, then switch it off in config
+      if (pulsar_config.has_key? 'tlsEnableHostnameVerification') &&
+          pulsar_config['tlsEnableHostnameVerification'].eql?('false')
+        client_config[:tls_validate_hostname] = false
+      end
+      # If 'TLS allow insecure connection' is false, then switch it off in config
+      if (pulsar_config.has_key? 'tlsAllowInsecureConnection') &&
+          pulsar_config['tlsAllowInsecureConnection'].eql?('false')
+        client_config[:tls_allow_insecure_connection] = false
+      end
+      # If token-based authentication is used, load the token
+      # Currently only loading from files is supported
+      if (pulsar_config['authPlugin'].eql? AUTH_TOKEN_PLUGIN_NAME) &&
+          pulsar_config['authParams'].start_with?(FILE_URL_PREFIX)
+        # read the first line from the token file
+        token = File.readlines(pulsar_config['authParams'].gsub(FILE_URL_PREFIX, ''))[0]
```

Comment:
We could add error handling for `Errno::ENOENT` here as well, just like in `read_config`.

Reply:
Thanks for the finding, I have fixed it in a commit below.